back/app/services/admin_publications_filtering_service.rb

```diff
@@ -48,6 +48,15 @@ class AdminPublicationsFilteringService
     scope
   end
 
+  add_filter('remove_childless_parents') do |scope, options|
+    next scope unless ['true', true, '1'].include? options[:remove_childless_parents]
+
+    parents_with_children = scope.where(id: scope.select(:parent_id).where.not(parent_id: nil).distinct)
+    non_parents           = scope.where(children_allowed: false)
```

Comment:
Are `non_parents` the same as `children` or are they something different semantically? If they are the same, I'd find `children` as a variable name more intention-revealing.

Reply:
`non_parents` and `children` are, indeed, different semantically: `children` are items with a parent (they are in a folder). `non-parents` includes items with no parent (e.g. top-level projects not in folders).